crates/ika-core/src/dwallet_mpc/make_dwallet_user_secret_key_shares_public.rs

```diff
@@ -0,0 +1,58 @@
+use anyhow::anyhow;
+use class_groups::Secp256k1EncryptionKey;
+use dwallet_mpc_types::dwallet_mpc::{
+    MPCPublicOutput, MPCPublicOutputClassGroups, SerializedWrappedMPCPublicOutput,
+};
+use group::{CyclicGroupElement, GroupElement};
+use homomorphic_encryption::{AdditivelyHomomorphicEncryptionKey, PlaintextSpaceGroupElement};
+use twopc_mpc::secp256k1::class_groups::AsyncProtocol;
+use twopc_mpc::secp256k1::SCALAR_LIMBS;
+
+/// Verifies the given secret share matches the given dWallets`
+/// DKG output centralized_party_public_key_share.
+pub fn verify_secret_share(
+    secret_share: Vec<u8>,
+    dkg_output: SerializedWrappedMPCPublicOutput,
+) -> anyhow::Result<bool> {
+    let secret_share: MPCPublicOutput = bcs::from_bytes(&secret_share)?;
+    let secret_share = match secret_share {
+        MPCPublicOutput::ClassGroups(MPCPublicOutputClassGroups::V1(output)) => output,
+        _ => {
+            return Err(anyhow!(
+                "invalid centralized public output version: expected ClassGroups::V1, got {:?}",
+                secret_share
+            ));
+        }
+    };
+    let dkg_output = bcs::from_bytes(&dkg_output)?;
+    match dkg_output {
+        MPCPublicOutput::ClassGroups(MPCPublicOutputClassGroups::V1(dkg_output)) => {
+            let expected_public_key =
+                cg_secp256k1_public_key_share_from_secret_share(secret_share)?;
+            let dkg_output: <AsyncProtocol as twopc_mpc::dkg::Protocol>::DecentralizedPartyDKGOutput =
+                bcs::from_bytes(&dkg_output)?;
+            Ok(dkg_output.centralized_party_public_key_share == expected_public_key.value())
```

Comment:
Use my PR

Reply:
Also use my pr for `verify_secret_share` in `crates/dwallet-mpc-centralized-party/src/lib.rs`